frontend/src/app/api/auth/[...nextauth]/config.ts

```diff
@@ -1,26 +1,63 @@
 import { AuthOptions } from "next-auth";
 import GoogleProvider from "next-auth/providers/google";
-
+import DiscordProvider from "next-auth/providers/discord";
 export const authOptions: AuthOptions = {
   providers: [
     GoogleProvider({
       clientId: process.env.NEXT_PUBLIC_GOOGLE_AUTH_CLIENT_ID as string,
       clientSecret: process.env.GOOGLE_AUTH_CLIENT_SECRET as string,
       checks: [],
     }),
+    DiscordProvider({
+      clientId: process.env.NEXT_PUBLIC_DISCORD_AUTH_CLIENT_ID as string,
+      clientSecret: process.env.DISCORD_AUTH_CLIENT_SECRET as string,
+    }),
   ],
   callbacks: {
     async jwt({ token, account }) {
       if (account) {
-        token.accessToken = account.access_token;
+        token.accessToken = account?.access_token;
+        token.provider = account?.provider;
+      } else {
+        console.warn("No account found");
       }
       return token;
     },
     async session({ session, token }) {
-      // @ts-ignore
       session.accessToken = token.accessToken;
+      session.provider = token.provider;
       return session;
     },
+    async redirect({ url: _url, baseUrl: _baseUrl }) {
+      const baseUrl = process.env.URL || _baseUrl;
+      const url = _url.startsWith("/")
+        ? `${baseUrl}${_url}`
+        : _url.replace(_baseUrl, baseUrl);
+
+      return url;
+    },
   },
   secret: process.env.NEXTAUTH_SECRET,
 };
+
+declare module "next-auth" {
```

Comment:
It's usually best practice to override/extend package types in a dedicated declaration file
for next-auth `next-auth.d.ts` in the root of the FE project